software/nosql/src/main/java/brooklyn/entity/nosql/riak/RiakNodeSshDriver.java

```diff
@@ -474,6 +475,18 @@ public void commitCluster() {
     }
 
     @Override
+    public void applyBucketType(String bucketTypeName, String bucketTypeProperties) {
```

Comment:
perhaps "createBucketType" would be a better method name, given that we all Riak's `bucket-type create ...`? (even though it means create or modify - still arguably better to be consistent with Riak's commands and documentation). Same goes for effector method on RiakNode as well (but the effector description can say "Create or modify a bucket type before activation" [1]).

Do we also need to activate the bucket type: "Like all bucket types, this type needs to be activated to be usable within the cluster." [1]. That is, call `riak-admin bucket-type activate ...`?

[1] http://docs.basho.com/riak/latest/dev/advanced/bucket-types/